### PR TITLE
fixed:#1887 Event and Holiday Calendar View

### LINKED ISSUE #1887: ADMIN & USER REDESIGN - List of events and holidays in calender view of events.
**Is your feature request related to a problem? Please describe.**
Currently, the event calendar does not provides the list of events and holidays.

**Describe the solution you'd like**

1. Adding a list of holidays, and
2. adding list events below the event calendar.
3. Using color coding for events

**Describe alternatives you've considered**
NA

**Additional context**

**Current**
Talawa-User
Screenshot from 2024-04-07 06-03-11
Talawa-Admin
Screenshot from 2024-04-07 05-45-06

We already figma design made for this.

Screenshot from 2024-04-07 05-46-34

Figma link: https://www.figma.com/file/dTGr5CnGHvkilU5nwnRXg2/Talawa-Admin-UI-Redesign?type=design&node-id=910-355&mode=design&t=EVDV5SZ2cKnOAc2R-0

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by gurramkarthiknetha**
This feature displays an interactive calendar view for tracking events and holidays. The calendar highlights holidays and user/organization-created events for the selected month, with color-coded markers for easy identification

**What kind of change does this PR introduce?**
This PR introduces a new feature for displaying an interactive event calendar with a list of holidays and user/organization-created events.

**Issue Number:**

Fixes #1887 

**Did you add tests for your changes?**
Yes, unit tests have been added to ensure the functionality works as expected.

**Snapshots/Videos:**
Snapshots and a short demonstration video showcasing the feature have been attached for better understanding.
image

**If relevant, did you update the documentation?**
Yes, the relevant sections in the [Talawa-Docs](https://github.com/PalisadoesFoundation/talawa-docs) have been updated to include details about the new feature.
**Summary**

This feature addresses the need for a more user-friendly and comprehensive event calendar.

Problem: The existing calendar lacked the ability to display a list of holidays and events in an organized manner.
Solution:
A list of holidays and events is now displayed below the calendar.
Color-coding has been implemented to distinguish between holidays and events for improved usability.
The UI aligns with the Figma design shared by the team.

**Does this PR introduce a breaking change?**

No, this PR does not introduce any breaking changes.

**Other information**

The Figma design used for this implementation can be found [here](https://www.figma.com/file/dTGr5CnGHvkilU5nwnRXg2/Talawa-Admin-UI-Redesign?type=design&node-id=910-355&mode=design&t=EVDV5SZ2cKnOAc2R-0).
This PR follows the project’s contribution guidelines and is submitted against the develop branch, as recommended.

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

## Summary by CodeRabbit

- **New Features**
	- Enhanced styling for the Event Calendar component, improving visual hierarchy and responsiveness.
	- Introduced new classes and CSS variables for better theme management and layout.
	- Improved rendering of holidays and events in the Event Calendar, with clearer separation and error handling.

- **Bug Fixes**
	- Adjusted logic in the Community Profile component for clearer control flow during profile data setting.

- **Chores**
	- Updated environment variables in the `.env.example` file for API and reCAPTCHA configurations.
	- Downgraded the `jest-preview` package version in `package.json`.

### CODE REVIEW

**Comment by Cioppolo14** [comment]
@gurramkarthiknetha Please fix the failed tests. Also, please do not add any reviewers.  Our PR review team will add reviewers when the PR is ready for a technical review.

**Comment by palisadoes** [comment]
Please make the code coverage of your patch reach 100%. It is currently at 48%

image

**Comment by palisadoes** [comment]
Please add valid test cases that were missed in the code . Please update it so that the codecov/patch exceeds 95.8%

- https://app.codecov.io/gh/PalisadoesFoundation/talawa-admin/pull/2633?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation

    image

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please add valid test cases that were missed in the code . Please update it so that the codecov/patch exceeds 95.8%

- https://app.codecov.io/gh/PalisadoesFoundation/talawa-admin/pull/2633?src=pr&el=tree&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=PalisadoesFoundation

    image

File: .env.example [hunk 1]
```diff
@@ -8,20 +8,20 @@ PORT=4321
 
 # Run Talawa-api locally in your system, and put its url into the same.
 
-REACT_APP_TALAWA_URL=
+REACT_APP_TALAWA_URL=http://localhost:4000/graphql/
```
Comment (palisadoes) [inline .env.example:11]: Your PR has too many unnecessary files that don’t relate to your issue. You can from the PR by running the commands below from the root directory of the repository

git add -u
git reset HEAD .env.example
git commit

Please apply these changes to this file.

**Review by palisadoes — CHANGES_REQUESTED** [review]
The following files must be excluded from the PR

1.  `.env.example`
2. `package.json`

They are not relevant to your changes

**Comment by palisadoes** [comment]
1. Please fix the failing tests. 
    1. You may have to fix a test that is not related to this PR as in some rare cases we merge faulty test code. If this is the case, fix the test in this PR. Do not open another issue.
1. Please make sure that CodeRabbit.ai approves the PR.

**Comment by gurramkarthiknetha** [comment]
> 1. Please fix the failing tests.
>    
>    1. You may have to fix a test that is not related to this PR as in some rare cases we merge faulty test code. If this is the case, fix the test in this PR. Do not open another issue.
> 2. Please make sure that CodeRabbit.ai approves the PR.

ya @palisadoes  1 test is failing that is not related to this pr 
it is related to the schema

**Comment by gurramkarthiknetha** [comment]
> 1. Please fix the failing tests.
>    
>    1. You may have to fix a test that is not related to this PR as in some rare cases we merge faulty test code. If this is the case, fix the test in this PR. Do not open another issue.
> 2. Please make sure that CodeRabbit.ai approves the PR.

what changes i need to do now

**Comment by palisadoes** [comment]
The Schema test is independent of the others and is expected to fail. We will create an issue to fix it later in the month.

Please fix the testing application test.

image

**Comment by gurramkarthiknetha** [comment]
> The Schema test is independent of the others and is expected to fail. We will create an issue to fix it later in the month.
> 
> Please fix the testing application test.
> 
> image

ok @palisadoes

**Comment by palisadoes** [comment]
Click on the Details links to see the cause of the problem. Please watch our `Getting Started - Developers` YouTube playlist.
- https://www.youtube.com/watch?v=dQuEKpsy1vM&list=PLv50qHwThlJUIzscg9a80a9-HmAlmUdCF&index=10

**Comment by palisadoes** [comment]
Thanks for submitting your PR, but we must close it.

1. We get notified of every event in the queue, and draft PRs like these, and their updates that require no action can be a distraction. We need our reviewers to focus on updates that are ready. 
1.  Don't be discouraged, we are just trying to keep our workloads to a manageable level.

Please resubmit when you are ready.

**Pull request closed without merging**